examples/squiral/main.go

```diff
@@ -0,0 +1,371 @@
+				if auto.colorMap[xtarg.x][xtarg.y] == s.col {
+					// If this is not free we cannot go into this direction.
+					set = false
+					break // try next direction
+				}
+			}
+
+			if set {
+				s.pos = target
+				s.dir = dir
+				// 2. set the color of this squiral to its
+				// current position.
+				setpix(s.pos, s.col)
+				return
+			}
+		}
+	}
+
+	s.dead = true
+}
+
+type automaton struct {
+	squirals [amount]squiral
+	colorMap [width][height]color.Color
+}
+
+func (au *automaton) init() {
+	// Init the test grid with color (0,0,0,0) and the borders of
+	// it with color(0,0,0,254) as a blocker color, so the squirals
+	// cannot escape the scene.
+	for x := 0; x < width; x++ {
+		for y := 0; y < height; y++ {
+			if x == 0 || x == width-1 || y == 0 || y == height-1 {
+				auto.colorMap[x][y] = blocker
+			} else {
+				auto.colorMap[x][y] = free
+			}
+		}
+	}
+
+	for i := 0; i < amount; i++ {
+		auto.squirals[i].spawn()
+	}
+}
+
+func (au *automaton) step() {
+	for i := 0; i < amount; i++ {
+		for s := 0; s < au.squirals[i].speed; s++ {
+			au.squirals[i].step(i)
+			if au.squirals[i].dead {
+				au.squirals[i].spawn()
+			}
+		}
+	}
+}
+
+func setpix(xy tup, col color.Color) {
+	canvas.Set(xy.x, xy.y, col)
+	auto.colorMap[xy.x][xy.y] = col
+}
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+	c, err := ebiten.NewImage(width, height, ebiten.FilterNearest)
+	if err != nil {
+		panic(err)
+	}
+	canvas = c
+	canvas.Fill(colornames.Black)
+
+	auto = &automaton{}
+	auto.init()
+}
+
+func update(screen *ebiten.Image) error {
+	reset := false
+
+	if ebiten.IsKeyPressed(ebiten.KeyEscape) {
+		return quit
+	}
+
+	if ebiten.IsKeyPressed(ebiten.KeyB) {
+		if background == colornames.White {
+			background = colornames.Black
+		} else {
+			background = colornames.White
+		}
+		reset = true
+	} else if ebiten.IsKeyPressed(ebiten.KeyT) {
+		selectedPalette = (selectedPalette + 1) % len(palettes)
+		reset = true
+	}
+
+	if ebiten.IsMouseButtonPressed(ebiten.MouseButtonLeft) {
+		reset = true
+	}
+
+	if reset {
+		screen.Fill(background)
+		canvas.Fill(background)
+		auto.init()
+	}
+
+	auto.step()
+
+	if ebiten.IsDrawingSkipped() {
+		return nil
+	}
+
+	draw(screen)
+
+	return nil
+}
+
+func draw(screen *ebiten.Image) {
+	op := &ebiten.DrawImageOptions{}
+	op.GeoM.Scale(drawScale, drawScale)
+	screen.DrawImage(canvas, op)
+	ebitenutil.DebugPrintAt(
+		screen,
+		fmt.Sprintf("TPS: %0.2f, FPS: %0.2f", ebiten.CurrentTPS(), ebiten.CurrentFPS()),
+		1, 1,
+	)
+	ebitenutil.DebugPrintAt(
+		screen,
+		"left mouse click: respawn",
+		1, 16,
+	)
+	ebitenutil.DebugPrintAt(
+		screen,
+		"b: toggle background (white/black)",
+		1, 32,
+	)
+	ebitenutil.DebugPrintAt(
+		screen,
+		fmt.Sprintf("t: cycle theme (current: %s)", palettes[selectedPalette].name),
+		1, 48,
+	)
+	ebitenutil.DebugPrintAt(
+		screen,
+		"esc: quit",
+		1, 64,
+	)
+}
+
+func main() {
+	ebiten.SetMaxTPS(250)
+	if err := ebiten.Run(update, wwidth, wheight, 2, "Squirals"); err != nil {
```

Comment:
Isn't this `Squirrels`? Or, is `Squirals` a common work for this kind of graphics?